EchoPro/survey.py

```diff
@@ -327,269 +328,502 @@ def read_validated_data( self ,
+                strata_floor = present_strata[present_strata < i]
+                strata_ceil = present_strata[present_strata > i]
+
+                new_stratum_below = np.max(strata_floor) if strata_floor.size > 0 else None
+                new_stratum_above = np.min(strata_ceil) if strata_ceil.size > 0 else None      
+                
+                sigma_bs_indexed = sigma_bs_impute[sigma_bs_impute['stratum_num'].isin([new_stratum_below, new_stratum_above])]
+                
+                sigma_bs_impute.loc[sigma_bs_impute.stratum_num==i , 'mean_sigma_bs'] = sigma_bs_indexed['mean_sigma_bs'].mean()
+                
+            self.acoustics['sigma_bs']['strata_mean'] = sigma_bs_impute        
+   
+    
+    def fit_binned_length_weight_relationship( self ,
+                                               species_id: np.float64 ):
+        """
+        Fit a length-weight relationship across discrete bins
+
+        Parameters
+        ----------
+        species_id : np.float64
+            Numeric code representing a particular species of interest
+
+        Notes
+        -----
+        This function first fits a length-weight regression based on measured 
+        values and then produces an array of fitted weight values (based on 
+        binned length values) that are used for later biomass calculations, 
+        weighting, and apportionment.
+        """    
+        
+        ### First make copies of each
+        specimen_df_spp = self.biology['specimen_df'].copy().pipe( lambda df: df.loc[ df.species_id == species_id ] )
+        length_weight_df = specimen_df_spp[['length', 'weight']].dropna(how='any')
+        
+        # pull distribution values
+        length_bins = self.biology['distributions']['length']['length_bins_arr']
+        length_intervals = self.biology['distributions']['length']['length_interval_arr']
+        
+        # length-weight regression
+        [ rate , initial ] = np.polyfit(np.log10(length_weight_df['length']), np.log10(length_weight_df['weight']), 1)
+        
+        # predict weight (fit equation)
+        fitted_weight = 10 ** initial * length_bins ** rate
+               
+        # add to object        
+        self.statistics['length_weight']['regression'] = {
+            'rate': rate ,
+            'initial': initial ,
+        }
+        
+        # summarize mean/samples of length and weight values (binned statistics)
+        length_bin_stats = length_weight_df.quantize_variables( bin_variable = 'length' ,
+                                                                bin_values = length_intervals )
+        
+        # fill bins where n_length < 5 w/ regressed weight values
+        length_bin_stats['modeled_weight'] = length_bin_stats['mean_weight'].copy()
+        low_n_indices = np.where(length_bin_stats['n_weight'].values < 5)[0].copy()   
+        length_bin_stats.loc[low_n_indices, 'modeled_weight'] = fitted_weight[low_n_indices].copy()
```

Comment:
This is just a style thing:
If I remember correctly, echopro variable names have been following the pattern of "noun_modifier", so `weight_mean`, `weight_n`, `weight_modeled` would better conform with that convention.